### ISSUE TITLE
Add streams analysis tab

### ISSUE THREAD

**Issue by ajrobbins** [WikiWatershed/model-my-watershed#2334]
Create a new Analysis tab called "streams", that displays summary statistics, by stream order and total, for the streams (NHD+) in  the AoI.

Here is a mockup of the tab's content: https://docs.google.com/document/d/1dEGFGqQLgvLmvV5NkzVszgqiJxYLrH5m1coEqg57q90/edit

Note that the "mean channel slope (%) statistic" represents length-weighted mean channel slope, and should be calculated as: sum(slope*length)/sum(length), where length is the segment attribute in NHD+.

**Comment by arottersman** [comment]
@ajrobbins the link in the description is taking me to a card about the user modal, which doesn't have the mockup for the stream tab content. Any idea where the actual wireframes are?

**Comment by ajrobbins** [comment]
Sorry about that @arottersman! Updated the issue description with the correct link.

**Comment by mmcfarland** [comment]
The referenced google doc indicates that slope is an attribute of  `nhdflowline`, but I'm not sure if that's captured in our table.  We may need to regenerate that data.  Given that, it seem like the stream order summary and the channel slope should be on separate cards.  I suspect that adding the panel and the new job just for streams is sufficient for this pointed work and the slope work might be shifted to a new card.

**Comment by kellyi** [comment]
I've been able to fit the tab by slightly adjusting the padding on the other tabs, but for the next tab we'll have to consider some way to page or toggle them:

<img width="517" alt="screen shot 2017-11-20 at 11 59 31 am" src="https://user-images.githubusercontent.com/4165523/33030864-78d74db6-cdea-11e7-8933-b29a0778b1b9.png">

**Comment by kellyi** [comment]
The range of stream order values in the `nhdflowline` table is currently 0-10, plus some which have `null` values. (Total row count for that table is 2993513; 325622 have `null` stream_order values, which is about 11%.)

How should I handle the `null` stream order rows? Discard them? Sum them together and send them back like the other values?

**Comment by mmcfarland** [comment]
Let's sum them and group them under "Other" while we determine what the null values actually indicate.

**Comment by kellyi** [comment]
Got the tab set up and set up the backend to return (currently-not-properly-formatted) data:

![screen shot 2017-11-20 at 6 08 43 pm](https://user-images.githubusercontent.com/4165523/33046213-08fd3c88-ce1e-11e7-9029-9a6116c824ac.png)

TODO:
- [ ] get stream length data formatted properly for client
- [ ] return data for all stream orders, including those represented by no streams in the aoi
- [ ] set up csv download
- [ ] add ag length and non-ag-length fields beneath the table

**Comment by kellyi** [comment]
> indicates that slope is an attribute of nhdflowline, but I'm not sure if that's captured in our table. We may need to regenerate that data

This is not currently available in the `nhdflowline` table, so I'm going to return 0 or null values for those fields from the backend for now and we can make commensurate API fixes when we update the table.

**Comment by kellyi** [comment]
Got the values returning properly from the database so we can show the stream lengths:

<img width="1020" alt="screen shot 2017-11-21 at 2 50 31 pm" src="https://user-images.githubusercontent.com/4165523/33093637-806ca706-cecb-11e7-80ed-c4bb24849f8c.png">

CSV download just works due to how we've set up the table.

Ag length and non-ag length aren't available fields in the db table, so I'm going to add their line items beneath the table but create another card for making the calculations.

**Comment by kellyi** [comment]
It also occurs to me that "0" may not be a meaningful value for stream order (seems like they run from 1st to 12th generally, but the range available in our db is 0-10), so if there are no objections I'm going to sum the 0 values with the `null` values and list them as `Other` in the table.

**Comment by arottersman** [comment]
Working on #2532, I saw that another flowline attribute (`slope`) was missing a similar number of values as `stream_order`. We've determined that this is to be expected because some of our flowlines are [canals, artificial paths, coast lines, etc](https://nhd.usgs.gov/userGuide/Robohelpfiles/NHD_User_Guide/Feature_Catalog/Hydrography_Dataset/Complete_FCode_List.htm) and don't have certain attributes.

@ajrobbins We might want to consult the client on whether they should be included or not — I'm not sure they classify as "Streams" if they're not of certain feature types (in the data `nhdflowline.ftype` and `nhdflowline.fcode`). Otherwise, "Other" does seem like an appropriate description of them.

-----------

Additional:
- All the feature types without attributes endup with `flowdir='Uninitialized'` (thanks @mmcfarland pointing out this category as something to look into) except for Coastlines.
- All null ordered flowlines have either `flowdir='Uninitialized'` or `ftype='Coastline'`.
- All 0 ordered flowlines have `ftype='Coastline'`.
